ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/toolbarview/MessengerToolbarFragmentTest.java

```diff
@@ -0,0 +1,570 @@
+    }
+
+    @Test
+    public void collapseToolbarViewToolbarExpanded() {
+        // Arrange
+        prepareMessengerToolbarFragment();
+        messengerToolbarFragment.configureForSimpleTitle(TEST_TEXT, true);
+
+        Whitebox.setInternalState(messengerToolbarFragment, LAST_ADDED_CHILD_FRAGMENT_FIELD, messengerToolbarExpandedFragment);
+
+        // Act
+        messengerToolbarFragment.collapseToolbarView();
+    }
+
+    @Test
+    public void isToolbarAreadyConfiguredWhenDefault() {
+        // Act
+        boolean toolbarAreadyConfigured = messengerToolbarFragment.isToolbarAreadyConfigured();
+
+        // Assert
+        assertFalse(toolbarAreadyConfigured);
+    }
+
+    @Test
+    public void refreshUnreadCounter() {
+        // Act
+        messengerToolbarFragment.refreshUnreadCounter(0);
+    }
+
+    @Test
+    public void removeAddedTransactedFragment() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+        // Arrange
+        prepareMessengerToolbarFragment();
+        messengerToolbarFragment.configureForSimpleTitle(TEST_TEXT, true);
+
+        Method removeAddedTransactedFragmentMethod =
+                messengerToolbarFragment.getClass().getDeclaredMethod(REMOVE_ADDED_TRANSACTED_FRAGMENT_METHOD);
+        removeAddedTransactedFragmentMethod.setAccessible(true);
+
+        // Act
+        removeAddedTransactedFragmentMethod.invoke(messengerToolbarFragment);
+    }
+
+    @Test
+    public void commitTransactionWithAnimationWhenLastActiveAgents() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+        // Arrange
+        prepareMessengerToolbarFragment();
+        messengerToolbarFragment.configureForLastActiveUsersView(lastActiveAgentsData, true);
+
+        Method commitTransactionWithAnimationMethod =
+                messengerToolbarFragment.getClass().getDeclaredMethod(COMMIT_TRANSACTION_WITH_ANIMATION_METHOD);
+        commitTransactionWithAnimationMethod.setAccessible(true);
+
+        // Act
+        commitTransactionWithAnimationMethod.invoke(messengerToolbarFragment);
+    }
+
+    @Test
+    public void commitTransactionWithAnimationWhenAssignedAgents() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+        // Arrange
+        prepareMessengerToolbarFragment();
+        messengerToolbarFragment.configureForAssignedAgentView(assignedAgentData, true);
+
+        Method commitTransactionWithAnimationMethod =
+                messengerToolbarFragment.getClass().getDeclaredMethod(MessengerToolbarFragmentTest.COMMIT_TRANSACTION_WITH_ANIMATION_METHOD);
+        commitTransactionWithAnimationMethod.setAccessible(true);
+
+        // Act
+        commitTransactionWithAnimationMethod.invoke(messengerToolbarFragment);
+    }
+
+    @Test
+    public void commitTransactionWithAnimationWhenSimpleTitle() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
```

Comment:
⚠️  By making sure that the test signature has `throws Exception`, the test becomes stable in case that the tested API changes and different exceptions are thrown.  

**[More details](https://confluence.devfactory.com/x/FBNrEw)**
          
## The possible solution in this case

```JAVA
 public void commitTransactionWithAnimationWhenSimpleTitle() throws Exception { 
   

```
 

<a id="690a6350-4f95-11e8-bb12-7de6652163a5" href="none" codereview-uuid-reserved-tag />